### PR TITLE
Adding Lambda Provisioned Concurrency Example with AutoScaling

### PR DESCRIPTION
**Pull request by Jacky-CC**
This example sets up Provisioned Concurrency for a Lambda function and configures autoscaling with target and schedule settings associated with Provisioned Concurrency.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: typescript/lambda-provisioned-concurrency-autoscaling/lib/lambda-pc-autoscaling-schedule.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  constructor(scope: Construct, id: string, props: LambdaPCScalingScheduleStackProps) {
+    super(scope, id, props);
+    this.lambdaFunctionName = props.functionName
+
+    // Create Sample Lambda Function which will create metrics
+    this.lambdaFunction = new NodejsFunction(this, 'LambdaFunction', {
+      functionName: this.lambdaFunctionName,
+      entry: `./lambda/lambda-handler.ts`,
+      runtime: Runtime.NODEJS_16_X,
```
Comment (kaizio) [inline typescript/lambda-provisioned-concurrency-autoscaling/lib/lambda-pc-autoscaling-schedule.ts:28]: Please update to 18_X.  Check issue #961 for more details on why. Lambda is simple to so it should not need more changes.

File: typescript/lambda-provisioned-concurrency-autoscaling/lib/lambda-pc-autoscaling-target.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  constructor(scope: Construct, id: string, props: LambdaPCScalingTargetStackProps) {
+    super(scope, id, props);
+    this.lambdaFunctionName = props.functionName
+
+    // Create Sample Lambda Function which will create metrics
+    this.lambdaFunction = new NodejsFunction(this, 'LambdaFunction', {
+      functionName: this.lambdaFunctionName,
+      entry: `./lambda/lambda-handler.ts`,
+      runtime: Runtime.NODEJS_16_X,
```
Comment (kaizio) [inline typescript/lambda-provisioned-concurrency-autoscaling/lib/lambda-pc-autoscaling-target.ts:29]: Same here

**Review by kaizio — COMMENTED** [review]
Requested some changes

File: typescript/lambda-provisioned-concurrency-autoscaling/lib/lambda-pc-autoscaling-schedule.ts (hunk 1 above)
Comment (Jacky-CC) [reply typescript/lambda-provisioned-concurrency-autoscaling/lib/lambda-pc-autoscaling-schedule.ts]: Thanks @kaizio, update done

**Review by kaizio — APPROVED** [review]
Thanks for the changes.

**Pull request merged**